homeassistant/components/flux_led/light.py

```diff
@@ -171,133 +141,231 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
+            )
+
+        async_add_entities(lights)
 
-    async_add_entities(lights)
+    await async_new_lights(entry.data[CONF_DEVICES])
 
+    async_dispatcher_connect(hass, SIGNAL_ADD_DEVICE, async_new_lights)
 
-class FluxLight(LightEntity):
-    """Representation of a Flux light."""
 
-    def __init__(self, device):
-        """Initialize the light."""
-        self._name = device["name"]
-        self._ipaddr = device[CONF_HOST]
-        self._protocol = device[CONF_PROTOCOL]
-        self._mode = device[ATTR_MODE]
-        self._custom_effect = device[CONF_CUSTOM_EFFECT]
-        self._effect_speed = device[CONF_EFFECT_SPEED]
-        self._bulb = None
-        self._error_reported = False
+class FluxLEDCoordinator(DataUpdateCoordinator):
+    """Update Coordinator for a specific light entity."""
 
-    def _connect(self):
-        """Connect to Flux light."""
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        name: str,
+        update_interval: int,
+        ip_address: str,
+        scan_coordinator: FluxLEDListUpdateCoordinator = None,
+    ):
+        """Initialize the update coordinator."""
+
+        super().__init__(
+            hass=hass,
+            logger=_LOGGER,
+            name=name,
+            update_interval=timedelta(milliseconds=update_interval),
+        )
 
-        self._bulb = WifiLedBulb(self._ipaddr, timeout=5)
-        if self._protocol:
-            self._bulb.setProtocol(self._protocol)
+        self._ip_address = ip_address
+        self._name = name
+        self.scan_coordinator = scan_coordinator
+
+        self.light = WifiLedBulb(self._ip_address)
+
+    async def _async_update_data(self):
+        """Fetch the data from this light bulb."""
+
+        await self.hass.async_add_executor_job(self.light.update_state)
+
+        get_rgbw = await self.hass.async_add_executor_job(self.light.getRgbw)
+        get_rgb = await self.hass.async_add_executor_job(self.light.getRgb)
+
+        light = {}
+
+        if self.light.mode == "ww":
+            light["mode"] = MODE_WHITE
+        elif self.light.rgbwcapable:
+            light["mode"] = MODE_RGBW
+        else:
+            light["mode"] = MODE_RGB
+
+        light["white_value"] = get_rgbw[3]
+
+        if light["mode"] == MODE_WHITE:
+            light["brightness"] = light["white_value"]
+        else:
+            light["brightness"] = self.light.brightness
+
+        light["hs_color"] = color_util.color_RGB_to_hs(*get_rgb)
+
+        light["current_effect"] = self.light.raw_state[3]
+
+        if self.light.is_on and light["brightness"] > 0:
+            light["state"] = True
+        else:
+            light["state"] = False
+
+        return light
+
+
+class FluxLight(CoordinatorEntity, LightEntity):
+    """Represents a Flux Light entity."""
+
+    def __init__(
+        self,
+        coordinator: FluxLEDCoordinator,
+        unique_id: str,
+        device: dict,
+        effect_speed: int,
+        hass: HomeAssistant,
```

Comment:
Don't pass in `hass`. It will be set on the entity when the entity has been added to home assistant. This means you can access `hass` with `self.hass`, after the entity has been added to home assistant.

Reply:
ok